hkm/finna.py

```diff
@@ -22,7 +22,44 @@ def __init__(self, organisation='"0/HKM/"', material_type='"0/Image/"'):
         self.organisation = organisation
         self.material_type = material_type
 
-    def search(self, search_term, page=1, limit=20, language='fi', detailed=False):
+    def get_facets(self, search_term, language='fi', date_from=None, date_to=None):
+        url = FinnaClient.API_ENDPOINT + 'search'
+        payload = {
+            'lookfor': search_term,
+            'filter[]': ['format:' + self.material_type, 'building:' + self.organisation, 'online_boolean:"1"'],
+            'limit': 0,
+            'lng': language,
+            'facet[]': ['author_facet', 'collection', 'genre_facet', 'main_date_str', 'category_str_mv']
+        }
+        if date_from or date_to:
+            payload['filter[]'].append('search_daterange_mv:"[%s TO %s]"' % (date_from, date_to))
```

Comment:
Finna UI seems to put an `*` to the date that is not set. I wonder if the search works correctly of the parameters is something like `search_daterange_mv:"[ TO 1980]`.

Quickly tested locally and it seems that we might get better/correct results if we use `*` as value of `date_from` or `date_to` in case either of them is empty/None.
I winder if we should also use `* ` Maybe we should replace the empty string or `None` with a `*`

Reply:
Seems that I totally missed the option of using `*` if value is empty. I'll test this locally as well and implement the change 👍